Gordon360/Controllers/RecIM/SeriesController.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Services.RecIM;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    [AllowAnonymous]
+    public class SeriesController : GordonControllerBase
+    {
+        private readonly ISeriesService _seriesService;
+
+        public SeriesController(ISeriesService seriesService)
+        {
+            _seriesService = seriesService;
+        }
+
+        /// <summary>
+        /// Queries all Series with an optional active tag
+        /// </summary>
+        /// <param name="active"></param>
+        /// <returns>Enumerable Set of Series</returns>
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<SeriesViewModel>> GetSeries([FromQuery] bool active)
+        {
+            var result = _seriesService.GetSeries(active);
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Gets specific Series
+        /// </summary>
+        /// <param name="seriesID"></param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("{seriesID}")]
+        public ActionResult<SeriesViewModel> GetSeriesByID(int seriesID)
+        {
+            var result = _seriesService.GetSeriesByID(seriesID);
+            return Ok(result);
+        }
+
+        [HttpPut]
+        [Route("")]
+        public async Task<ActionResult> UpdateSeries(UpdateSeriesViewModel s)
```

Comment:
Traditionally in a REST API, you would PUT data to an endpoint with an id at the end. Therefore, you pass a unique "name" of a resource and the data that you want to place at that location.
```suggestion
        [Route("{seriesID}")]
        public async Task<ActionResult> UpdateSeries(int seriesID, [FromBody] UpdateSeriesViewModel s)
```
Also, you will need to specify frombody for the model since it will otherwise look for it in the URL GET parameters.

Reply:
> Also, you will need to specify frombody for the model since it will otherwise look for it in the URL GET parameters.

This is not true, [according to the docs](https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-6.0#sources-1) ASP.NET Core will attempt to bind models in this order:

1. Form fields
2. The request body (For [controllers that have the [ApiController] attribute](https://learn.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-6.0#binding-source-parameter-inference).)
3. Route data
4. Query string parameters
5. Uploaded files